torch/csrc/jit/tensorexpr/codegen.h

```diff
@@ -121,44 +131,34 @@ class CodeGen::CallArg {
 
   template <typename T>
   // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init,cppcoreguidelines-pro-type-const-cast)
-  CallArg(const std::vector<T>& buffer) : ptr_(const_cast<T*>(buffer.data())) {}
+  CallArg(const std::vector<T>& buffer)
+      : data_(const_cast<T*>(buffer.data())) {}
 
   // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
-  CallArg(void* ptr) : ptr_(ptr) {}
+  CallArg(void* ptr) : data_(ptr) {}
 
-#define ARG_TYPE_CTOR(Type, Name) \
-  CallArg(Type v) : Name##val_(v) {}
+#define ARG_TYPE_CTOR(Type, Name)     \
+  CallArg(Type v) {                   \
+    memcpy(&data_, &v, sizeof(Type)); \
+  }
   // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
   AT_FORALL_SCALAR_TYPES_AND2(Bool, Half, ARG_TYPE_CTOR);
```

Comment:
This doesn't included complex, does it?  As long as all the scalar types fit in a void* this is ok.